lib/ui/painting/image_shader.cc

```diff
@@ -45,7 +45,7 @@ void ImageShader::initWithImage(CanvasImage* image,
         ToDart("ImageShader constructor called with non-genuine Image."));
     return;
   }
-  sk_image_ = image->image();
+  sk_image_ = UIDartState::CreateGPUObject(image->image());
```

Comment:
This is an example of the multiple guard objects that I was referring to in a recent discussion. The SkiaGPUObject held by CanvasImage is a unique pointer and so we have to construct a new guard object here. If we made it shared with ref counts then we could use the same object here and elsewhere. The downside of the unique guard objects is that we have to enqueue both for an unref on the raster thread and only one of those unrefs will actually delete the object.

That's not a problem for this PR, but it's an open issue to consider. Hopefully if Skia makes SkImage thread-safe we can ignore all of this...